src/main/scala/com/databricks/spark/redshift/RedshiftRelation.scala

```diff
@@ -98,12 +98,13 @@ private[redshift] case class RedshiftRelation(
   }
 
   protected def compileValue(value: Any): Any = value match {
-    case stringValue: String => s"\\'${escapeSql(stringValue.toString)}\\'"
-    case _ => value
+    case stringValue: String => s"\\'${escapeSql(stringValue)}\\'"
+    case stringValue: UTF8String => s"\\'${escapeSql(stringValue.toString())}\\'"
```

Comment:
This is a Spark bug, since we shouldn't ever be exposing `UTF8String` to third-party code. We can avoid directly relying on UTF8String if we use the schema to look up column types instead. I'm going to perform this refactoring now.